pkg/operation/botanist/controlplane.go

```diff
@@ -600,3 +594,79 @@ func (b *Botanist) DeleteCertBroker() error {
 	}
 	return nil
 }
+
+// CreateDeletionCheckpoint creates an empty configmap used as a checkpoint for shoot deletion. It gets created after the Kubernetes resources
+// inside a shoot have been cleaned up and the kube-apiserver is deleted.
+func (b *Botanist) CreateDeletionCheckpoint() error {
+	_, err := b.K8sSeedClient.CreateConfigMap(b.Shoot.SeedNamespace, common.DeletionCheckpoint, nil, true)
+	if apierrors.IsAlreadyExists(err) {
+		return nil
+	}
+	return err
+}
+
+// WakeUpControlPlane scales the replicas to 1 for the following deployments which are needed in case of shoot deletion:
+// * etcd-events
+// * etcd-main
+// * kube-apiserver
+// * cloud-controller-manager
+// * kube-controller-manager
+// * machine-controller-manager
+func (b *Botanist) WakeUpControlPlane() error {
```

Comment:
Regarding all control plane components: What if the shoot is also a seed? In this case, the shoot would have to be scaled again afterwards since the replica counts are different...

Reply:
What do you mean?